flask/app/analyses.py

```diff
@@ -173,10 +174,34 @@ def create_analysis():
     requested = now
     updated = now
     analysis_state = "Requested"
-    try:
-        requester_id = updated_by_id = current_user.user_id
-    except:  # LOGIN DISABLED
-        requester_id = updated_by_id = 1
+
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+    else:
+        user_id = current_user.user_id
+
+    requester_id = updated_by_id = current_user.user_id
```

Comment:
Okay, I think here we need a solution for the logic
- LOGIN_DISABLED -> must specify ?user= to populate this
- admin -> use ?user= and fallback to the admin's real ID
- regular -> real ID

Reply:
Something along the lines of this? I'm not sure I understand what the fallback value (real ID) is exactly, see the ??? below
```
    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
        user_id = requester_id = updated_by_id = request.args.get(
            "user", ???
        )
    else: # reg user
        user_id = requester_id = updated_by_id = current_user.user_id
```

Reply:
Not quite as we currently depend on user_id being None to determine that the user is an admin and not restrict the query on datasets

Reply:
```python
if not user_id and app.config.get("LOGIN_DISABLED"):
    requester_id = updated_by_id = 1
else:
    requester_id = updated_by_id = user_id or current_user.user_id
```
I _think_ this solves the problem

Reply:
for my own understanding, since user_id is evaluated to determine if someone is an admin (for permission purposes) - what are the cases where it is assigned None? 
```
    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
        user_id = request.args.get("user") # assigned None here if no user query parameter is specified b/c of the get method 
    else:
        user_id = current_user.user_id # this will always return a value right because this would only be evaluated if the user is not an admin and therefore a regular user?
```
and then the check you have up there is basically accounting for that case where user_id is None, as a result of it not being specified as a query parameter. does this sound correct to you @kevinlul

Reply:
None means either LOGIN_DISABLED or is_admin but didn't provide a `?user=` to masquerade as